pkg/dexidp/provider.go

```diff
@@ -83,7 +107,7 @@ func (p *dexProvider) Configure(ctx context.Context, req provider.ConfigureReque
 		return
 	}
 
-	client, err := client.New(host)
+	client, err := constructDexClient(host, config.TLS)
```

Comment:
```suggestion
	client, err := newDexClient(host, config.TLS)
```

To be a bit more golang naming conventions